simulation_ws/src/sagemaker_rl_agent/markov/single_machine_training_worker.py

```diff
@@ -40,6 +41,16 @@ def add_items_to_dict(target_dict, source_dict):
 def should_stop_training_based_on_evaluation():
     return False
 
+class SigTermHandler:
+    def __init__(self, data_store):
+        self.data_store = data_store
+        # signal.signal(signal.SIGINT, self.sigterm_exit)
+        signal.signal(signal.SIGTERM, self.sigterm_exit)
+
+    def sigterm_exit(self, signum, frame):
+        self.data_store.unlock()
```

Comment:
Does this fail if the lock file does not exist?

Reply:
It should be the same behavior as [here](https://github.com/aws-robotics/aws-robomaker-sample-application-objecttracker/pull/29/files#diff-ef98845f53fd54190691649a67f557a4R71), which is it does not fail.

Reply:
This behaviour seems fine and fixes a bug, so I will approve. It may be worthwhile to write tests for some of this behaviour so that it can easily be verified to work